analyzers/tests/SonarAnalyzer.UnitTest/TestCases/NativeMethodsShouldBeWrapped.CSharp11.SourceGenerator.cs

```diff
@@ -0,0 +1,86 @@
+﻿// <auto-generated/>
+namespace LibraryImportAttributeTests
+{
```

Comment:
can you also add a "trivial wrapper" here , to test `ReportTrivialWrappers` isn't triggered on generated code?

Reply:
Someone might not be aware of your initial intention and manually change the file in the future.

What do you suggest to avoid this situation? Can this process be automated and part of the test execution?

Reply:
> Can this process be automated and part of the test execution?

I wish we could but this is pretty difficult to achieve. The source generators are itself based on analyzers (The ImportLibraryGenerator.dll gets passed to csc.exe via the /analyzer switch). If we want to enable our test infrastructure to support this (which is desirable in the long run), we need to dedicate a sprint to it.

> Or add a comment at the top of the file, at least?

On top doesn't work because of the `<auto-generated>` comment. I was thinking about a comment too, but this violates the copy/paste idea also. if we add a comment, we need to add it to the test method. I could use my bullet point list from above as a template. What do you think about this?

Reply:
yes, a comment above the test method sounds good to me